### DISCUSSION TITLE
How to rename a folder and image in a repo

### DISCUSSION THREAD

**Discussion by NitinIsFit** [community/community#28271] [Other Feature Feedback, Questions, & Ideas] [answered]
How to rename a folder in your repository?

![Screenshot_20220804-060954.png](https://user-images.githubusercontent.com/94877880/182738867-08347365-696f-4bc0-8216-7a265ded6ed9.png)

**Comment by NitinIsFit** [comment]
We can't even rename an image

![Screenshot_20220804-061323.png](https://user-images.githubusercontent.com/94877880/182738987-ff66f282-e542-4960-891d-667176892e20.png)

**Comment by jfullstackdev** [comment]
I think you already renamed the folder ? 

as I was checking, it's already small letters

![image](https://user-images.githubusercontent.com/47092464/182744734-b7ecacfc-982d-480c-b394-8e3c6029af1d.png)

but for images, since it's being treated as binary and it's just an add-on, because as we all know, GitHub is mainly for code, by default, you can't

but GitHub Desktop can since it will treat it as local files once cloned

![image](https://user-images.githubusercontent.com/47092464/182745032-5d465054-2327-4dc7-a12e-85fc38e7895d.png)

**Reply by NitinIsFit** [reply]
> I think you already renamed the folder ? 
> 
> as I was checking, it's already small letters
> 
> ![image](https://user-images.githubusercontent.com/47092464/182744734-b7ecacfc-982d-480c-b394-8e3c6029af1d.png)
> 
> but for images, since it's being treated as binary and it's just an add-on, because as we all know, GitHub is mainly for code, by default, you can't
> 
> but GitHub Desktop can since it will treat it as local files once cloned
> 
> ![image](https://user-images.githubusercontent.com/47092464/182745032-5d465054-2327-4dc7-a12e-85fc38e7895d.png)
> 

I did it by making a new folder, deleting the files, and re-uploading in the new folder.

When i rename the `Images/test.txt` as `images/text.txt` it doesn't rename the existing folder, but it makes a new one and move the file there.

**Reply by NitinIsFit** [reply]
If you have to rename multiple files have to:

`Images/image1.png`→`images/image1.png`
`Images/image2.png`→`images/image2.png`
`Images/image3.png`→`images/image3.png`
`Images/image4.png`→`images/image4.png`

And so on...
Bruh

**Reply by NitinIsFit** [reply]
Well, and we cannot rename a image... 😐

**Reply by NitinIsFit** [reply]
What does that popup menu say in the bottom of the screen? It says "you cannot create a codespace in the..." In the what? It doesn't appear on the screen properly.
There is so much confusion going on.

![Screenshot_20220805-211458.png](https://user-images.githubusercontent.com/94877880/183113363-8bb94e9c-24f9-4eaf-8d6f-95c64800a5b5.png)

**Comment by airtower-luna** [comment, marked as answer]
If you use Git on the command line, it's as easy as:

```
git mv Images images
git commit -m "Rename Images/ to images/"
```

I hear on Windows there are sometimes issues with that because it's case-insensitive filesystem considers the names the same. In that case you could do the rename in two steps, like rename `Images` to `images1` and then from that to `images`. :sweat_smile:

**Reply by NitinIsFit** [reply]
Wish the site had this function.

**Comment by intgswd** [comment]
You still can't rename an image from the browser interface, although you can rename code-based files. I would love it if this was possible.

**Comment by NokiasSOFT** [comment]
That should be possible, as well as being able to rename any file without having to edit it so you don't have to work around it to not mess up your original encoding.
